check_pgactivity

```diff
@@ -5680,76 +5687,145 @@ sub check_replication_slots {
+                    $warn{$threshold.'_files'} = $value;
+              }
+              while ( $args{'critical'} =~ /$threshods_re/g ) {
+                  my ($threshold, $value) = ($1, $2);
+                    $crit{$threshold.'_files'} = $value;
+              }
 
-            if ( $row->[1] >= $args{'critical'} ) {
-                push @msg_crit => "$row->[0]: $row->[1] WAL";
-                next SLOTS_LOOP;
             }
+          # For backward compatibility
+          elsif ($args{'warning'}  =~ m/^([0-9]+)$/
+                 and $args{'critical'} =~ m/^([0-9]+)$/) {
+                   $warn{wal_files} = $args{'warning'};
+                   $crit{wal_files} = $args{'critical'};
+                 }
+          else {
+            pod2usage(
+                -message => "FATAL: critical and warning thresholds only accept:\n"
+                    . "- raw numbers for backward compatibility to set wal threshold.\n"
+                    . "- a list 'wal=value' and/or 'replslot=value' separated by comma.\n"
+                    . "See documentation for more information.",
+                -exitval => 127
+            )
+          }
 
-            if ( $row->[1] >= $args{'warning'} ) {
-                push @msg_warn => "$row->[0]: $row->[1] WAL";
-            }
+        dprint ('replication_slots_warn: '. Dumper(%warn));
+        dprint ('replication_slots_crit: '. Dumper(%crit));
+
+    }
+
+    @perf_wal_limits = ( $warn{wal_files}, $crit{wal_files} ) if defined $warn{wal_files};
+    @perf_replslot_limits = ( $warn{replslot_files}, $crit{replslot_files} ) if defined $warn{replslot_files};
+
+
+    @rs = @{ query_ver( $hosts[0], %queries ) };
+
+SLOTS_LOOP: foreach my $row (@rs) {
+        push @perfdata => [ "$row->[0]_wal", $row->[2],'File', @perf_wal_limits ];
+        push @perfdata => [ "$row->[0]_replslots", $row->[3], 'File', @perf_replslot_limits ];
+
+        if ( defined $crit{wal_files}) {
+          if ( $row->[2] > $crit{wal_files} ) {
+              push @msg_crit, "$row->[0] wal files : $row->[2]";
+              push @longmsg => sprintf("Slot: %s wal files = %s above crit threshold %s",
+                  $row->[0], $row->[2], $crit{wal_files}
+              );
+          }
+          elsif ( $row->[2] > $warn{wal_files} ) {
+                push @msg_warn, "$row->[0] wal files : $row->[2]";
+                push @longmsg => sprintf("Slot: %s wal files = %s above warn threshold %s",
+                    $row->[0], $row->[2], $warn{wal_files}
+                );
+          }
+        }
+        if ( defined $crit{replslot_files}) {
+          if ( $row->[3] > $crit{replslot_files} ) {
+              push @msg_crit, "$row->[0] pg_replslot files : $row->[3]";
+              push @longmsg => sprintf("Slot: %s pg_replslot files = %s above crit threshold %s",
+                  $row->[0], $row->[3], $crit{replslot_files}
+              );
+          }
+          elsif ( $row->[3] > $warn{replslot_files} ) {
```

Comment:
Same problem as above: fails if "replslot=" was defined only for Critical:
```
check_pgactivity --service replication_slots --dbservice nagios_pagode10 --debug -c 'replslot=1' -w 'wal=99'

Use of uninitialized value in numeric gt (>) at /opt/check_pgactivity_anayrat/check_pgactivity line 5812.
Use of uninitialized value in sprintf at /opt/check_pgactivity_anayrat/check_pgactivity line 5816.
Use of uninitialized value in numeric gt (>) at /opt/check_pgactivity_anayrat/check_pgactivity line 5812.
Use of uninitialized value in sprintf at /opt/check_pgactivity_anayrat/check_pgactivity line 5816.
```

Reply:
Should be fixed